### DISCUSSION TITLE
JSON schema for declarative configuration?

### DISCUSSION THREAD

**Discussion by ghost** [Kong/kong#9094] [Help]
Hello, just wondering if there exists a JSON schema for the declarative configuration?

**Comment by ghost** [comment]
I see [here](https://docs.konghq.com/gateway/latest/admin-api/#retrieve-entity-schema) that you can get _a schema_ from `kong:8001/schemas/{entityName}`.

I am not sure what schema format that is, but at least it's something that can probably mostly be transformed into a JSON schema.

**Comment by hanshuebner** [comment]
> I am not sure what schema format that is, but at least it's something that can probably mostly be transformed into a JSON schema.

The format reported by the `/schemas/:entityName` endpoint resembles the internal schema format used by Kong gateway to describe entities stored in the configuration database.  It is documented in the [Plugin Development Documentation](https://docs.konghq.com/gateway/latest/plugin-development/custom-entities/#define-a-schema).  You may want to have a look at the [definition of the core entities](https://github.com/Kong/kong/tree/2.8.0/kong/db/schema/entities) to see how the JSON representation reported by the admin API corresponds to the actual definitions.

Kong gateway does not currently have a way to report schemas in JSON Schema format.
